src/MatrixFields/field_name_dict.jl

```diff
@@ -237,6 +287,103 @@ function Base.one(matrix::FieldMatrix)
     return FieldNameDict(inferred_diagonal_keys, entries)
 end
 
+"""
+    get_field_first_index_offset(name::FieldName, ::Type{T}, ::Type{S})
+
+Returns the offset of the the field with name `name` in an object of type `S`
+in multiples of `sizeof(T)`.
+"""
+function get_field_first_index_offset(
+    name::FieldName,
+    ::Type{T},
+    ::Type{S},
+) where {T, S}
+    if name == @name()
+        return 0
+    end
+    child_name = extract_first(name)
+    child_type = fieldtype(S, child_name)
+    remaining_field_chain = drop_first(name)
+    field_index =
+        unrolled_filter(i -> fieldname(S, i) == child_name, 1:fieldcount(S))[1]
+    return DataLayouts.fieldtypeoffset(T, S, field_index) +
+           get_field_first_index_offset(remaining_field_chain, T, child_type)
+end
+if hasfield(Method, :recursion_relation)
+    dont_limit = (args...) -> true
+    for m in methods(get_field_first_index_offset)
+        m.recursion_relation = dont_limit
+    end
+end
+
+"""
+    get_scalar_keys(dict::FieldMatrix)
+
+Returns a `FieldMatrixKeys` object that contains the keys of all the scalar
+entries in the `FieldMatrix` `dict`.
+"""
+function get_scalar_keys(dict::FieldMatrix)
+    keys_tuple = unrolled_flatmap(keys(dict).values) do key
+        entry = dict[unrolled_filter(isequal(key), keys(dict).values)[1]]
+        entry =
+            entry isa ColumnwiseBandMatrixField ? entry.entries.:(1) : entry
+        unrolled_map(
+            filtered_names(entry) do field
+                if field isa UniformScaling
+                    return true
+                elseif field isa Fields.Field
+                    return eltype(field) == eltype(eltype(field))
+                else
+                    return eltype(field) == typeof(field)
+                end
+            end,
```

Comment:
```suggestion
            filtered_names(entry) do field
                if field isa UniformScaling
                    true
                elseif field isa Fields.Field
                    eltype(field) == eltype(eltype(field))
                else
                    eltype(field) == typeof(field)
                end
            end,
```
I think using `return` inside `filter` / `map` is confusing here.